fili-core/src/main/java/com/yahoo/bard/webservice/logging/TimedPhase.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a phase that is timed.
+ * TimedPhase is used to associate a Timer located in the registry with the exact duration of such a phase for a
+ * specific request.
+ */
+public class TimedPhase implements AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(TimedPhase.class);
+
+    private final String name;
+    private long start;
+    private long duration;
```

Comment:
Actually, after playing with it a little bit, I think I'm going to punt on this, largely because validating whether a timer is running outside of this class or not becomes a lot trickier in a multi-threaded scenario.